app/scripts/filters/util.js

```diff
@@ -528,6 +528,36 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(envVar) {
+      if(envVar.value) {
+        return;
+      }
+      if(!envVar.valueFrom) {
+        return;
+      }
+      envVar.valueIcon = 'pficon pficon-help';
+      envVar.valueIconTooltip = 'This is a referenced value that will be generated when a container is created.  On running pods you can check the resolved values by going to the Terminal tab and echoing the environment variable.';
+      _.each(envVar.valueFrom, function(from, refType) {
+        switch(refType) {
+          case 'configMapKeyRef':
+            envVar.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
```

Comment:
Updated to use [a util function](https://github.com/openshift/angular-key-value-editor/blob/master/src/services/key-value-editor-utils.js#L98) to clean up the entries before saving.  The fn deletes only "known keys" to the kve, and will drop an entry if it has no key or value (since its possible to do that the way it creates new rows).

@jwforres @spadgett